modules/user/user.repository.js

```diff
@@ -1,35 +1,84 @@
 const mongoose = require("mongoose");
 const Schema = mongoose.Schema;
 const { CustomError } = require("../../errorHandler");
-require("dotenv").config();
+const { encodeJWT } = require("./jwt.config");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
 
 const userScheme = new Schema({
-    id: Number,
-    name: String,
-    surname: String,
-    email: String,
-    phone: Number,
+    id: {
+        type: Number,
+        required: true,
+        unique: true,
+    },
+    name: {
+        type: String,
+        required: true,
+    },
+    surname: {
+        type: String,
+        required: true,
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    phone: {
+        type: Number,
+        required: true,
+        unique: true,
+    },
+    password: {
+        type: String,
+        required: true,
+    },
 });
 const UserModel = mongoose.model("User", userScheme);
 
 const userAggregateForm = {
-    _id: 0,
-    id: 1,
-    name: 1,
-    surname: 1,
-    email: 1,
-    phone: 1,
+    _id: false,
+    id: true,
+    name: true,
+    surname: true,
+    email: true,
+    phone: true,
+    password: true,
 };
 
 class UserRepository {
+    async userLogin(userCred) {
```

Comment:
this method should be located in `user.service.js` cause here we can see logic process about user login like comparing of passwords and etc. From DAL here only getting user from database. 
Imagine that we need to change DB, logic about authentication shouldn't be changed in this case.

- [ ] move business logic to server